File: api/app/Models/Customer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Customer extends Model
{
    use SoftDeletes;
    protected $fillable = [
        'full_name', 'dni', 'phone', 'email',
        'province_id', 'district_id', 'address', 'is_active',
        // Campos fiscales
        'document_type', 'ruc', 'razon_social', 'nombre_comercial',
    ];

    protected $casts = [
        'is_active' => 'boolean',
    ];

    public function province(): BelongsTo
    {
        return $this->belongsTo(Province::class, 'province_id');
    }

    public function district(): BelongsTo
    {
        return $this->belongsTo(District::class, 'district_id');
    }

    public function orders(): HasMany
    {
        return $this->hasMany(Order::class, 'customer_id');
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Catalog/Entities/Customer.cs
using AdminHiitop.Api.Domain.Common;
using AdminHiitop.Api.Domain.Sales.Entities;

namespace AdminHiitop.Api.Domain.Catalog.Entities;

public sealed class Customer : AuditableEntity
{
    public string FullName { get; set; } = string.Empty;
    public string? Dni { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public int? ProvinceId { get; set; }
    public int? DistrictId { get; set; }
    public string? Address { get; set; }
    public bool IsActive { get; set; } = true;
    public string? DocumentType { get; set; }
    public string? Ruc { get; set; }
    public string? RazonSocial { get; set; }
    public string? NombreComercial { get; set; }

    public Province? Province { get; set; }
    public District? District { get; set; }
    public ICollection<Order> Orders { get; set; } = new List<Order>();
}
